password/src/Data/Password/PBKDF2.hs

```diff
@@ -249,12 +258,27 @@ checkPassword pass (PasswordHash passHash) =
     salt <- from64 salt64
     hashedKey <- from64 hashedKey64
     let pbkdf2OutputLength = fromIntegral $ C8.length hashedKey
-        producedKey = hashPasswordWithSalt' PBKDF2Params{..} (Salt salt) pass
-    guard $ hashedKey `constEq` producedKey
-    return PasswordCheckSuccess
+    return (PBKDF2Params{..}, Salt salt, hashedKey)
   where
     pbkdf2Salt = 16
 
+-- | Extract 'PBKDF2Params' from a 'PasswordHash' 'PBKDF2'.
+--
+-- Returns 'Just PBKDF2Params' on success.
+--
+-- (Note that 'argon2Salt' is defaulted)
```

Comment:
`pbkdf2Salt`, though you don't have to default it, I'm pretty sure that `Data.ByteString.length salt` will give you the initially used parameter.

Same with `scryptSalt` and `argon2Salt`.